engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -27,22 +34,26 @@ public class ProcessInstanceMigrationMigrateProcessor
   private final TypedResponseWriter responseWriter;
   private final ElementInstanceState elementInstanceState;
   private final ProcessState processState;
+  private final JobState jobState;
 
   public ProcessInstanceMigrationMigrateProcessor(
       final Writers writers,
       final ElementInstanceState elementInstanceState,
-      final ProcessState processState) {
+      final ProcessState processState,
+      final JobState jobState) {
     stateWriter = writers.state();
     responseWriter = writers.response();
     this.elementInstanceState = elementInstanceState;
     this.processState = processState;
+    this.jobState = jobState;
   }
 
   @Override
   public void processRecord(final TypedRecord<ProcessInstanceMigrationRecord> command) {
     final ProcessInstanceMigrationRecord value = command.getValue();
     final long processInstanceKey = value.getProcessInstanceKey();
     final long targetProcessDefinitionKey = value.getTargetProcessDefinitionKey();
+    final var mappingInstructions = value.getMappingInstructions();
 
     final ElementInstance processInstance = elementInstanceState.getInstance(processInstanceKey);
     if (processInstance == null) {
```

Comment:
🔧 Now, as the method is growing, it would be good if we extract both preconditions into methods e.g. `ensureProcessInstanceExist()`.